### ISSUE TITLE
npm ERR! fetch failed https://s3.amazonaws.com/tool/tool-0.1.0.tgz (v2.13.5)

### ISSUE THREAD

**Issue by AshCoolman** [npm/npm#9653]
Its been a while since I `npm i`'d but I vaugely remember upgrading npm a few weeks ago. I'm unsure what version npm was, but now its `v2.13.5`

My previously working, now breaking package json was:

```
  "dependencies": {
    "tool": "http://s3.amazonaws.com/tool/tool-0.1.0.tgz"
  },
```

which causes this error:

```
npm ERR! fetch failed https://s3.amazonaws.com/tool/tool-0.1.0.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 400
```

`$ CURL http://s3.amazonaws.com/tool/tool-0.1.0.tgz` works fine

Hm, just tried with `v2.11.3` and `v2.10.0`. Perhaps its my network or something.

**Comment by kirstein** [comment]
What was the issue here? Im having similar problem currently. Strange thing is that its only failing on a certain computer.

npm: 2.14.7
node: v4.2.1
s3 log: 
```
  2348D9998130BEB0 REST.GET.OBJECT packaging/node/platform-0.0.1.tgz "GET   burp/packaging/node/platform-0.0.1.tgz HTTP/1.1" 400 InvalidArgument 379 - 6 - "install" "npm/2.14.7 node/v4.2.1 darwin x64" -
```

**Comment by AshCoolman** [comment]
I actually solved the problem back when I posted this message. But just 2 days ago this started happening again and I have totally forgotten what happened - though I do remember trying all sorts of things. But this time I gave up and integrated the module into my repo source. When I get some time I will try have another crack.

**Comment by AshCoolman** [comment]
I think I fixed it by downgradeding `npm i -g npm@1.4.21`

**Comment by kirstein** [comment]
you are right, that seems to solve it!

**Comment by AshCoolman** [comment]
Just a note, this is still a problem for me with 3.5.1

**Comment by lukemelia** [comment]
I had this problem too on 3.7.1, and found that `npm logout` clears it up. If I `npm login`, the problem returns.

**Comment by AshCoolman** [comment]
Question @lukemelia & @kirstein , what OS are you guys on? (I'm on OSX)

**Comment by mattbailey** [comment]
Having this issue too, npm is trying to authenticate with a challenge from S3.  Since S3 does, in fact, use bearer token auth like npmjs.com does, this fails.

```
20 http fetch GET https://[redacted]
21 verbose request using bearer token for auth
22 verbose request id 7efec7d8a53d65da
23 http fetch 400 https://[redacted]
24 error fetch failed https://[redacted]
25 warn retry will retry, error on last attempt: Error: fetch failed with status code 400
```

`npm logout` resolves this, but pretty annoying

**Comment by KyleCartmell** [comment]
seems to be fixed in 3.10.6

**Comment by KenanY** [comment]
Closing as resolved then :)
